example/src/androidTestDebug/java/com/stripe/example/activity/CreateCardTokenActivityTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.stripe.example.activity
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.onData
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.rules.activityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import com.stripe.example.R
+import org.hamcrest.CoreMatchers.anything
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class CreateCardTokenActivityTest {
+
+    @get:Rule
+    val activityScenarioRule: ActivityScenarioRule<LauncherActivity> = activityScenarioRule()
+
+    @Before
+    fun setup() {
+        Intents.init()
+    }
+
+    @After
+    fun intentsTeardown() {
+        Intents.release()
+    }
+
+    @Test
+    fun createCardToken() {
+        // launch create a card token activity
+        onView(withId(R.id.examples)).perform(
+            RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(1, click())
+        )
+
+        // fill out card details
+        onView(withId(R.id.et_card_number))
+            .perform(typeText("4242424242424242"))
+        onView(withId(R.id.et_expiry_date))
+            .perform(typeText("01/25"))
+        onView(withId(R.id.et_cvc))
+            .perform(typeText("111"))
+
+        // click create card button
+        onView(withId(R.id.create_token_rxjava))
+            .perform(click())
+
+        // don't use Thread.sleep in Espresso tests - figure out how to use IdlingResource
+        Thread.sleep(2000)
```

Comment:
Format as a `TODO` to make this more discoverable? Unless you're coming right back to this

Reply:
Good idea. I have a followup that touches this file so I'll add there.